crates/fuel-core/src/service/adapters/consensus_module/poa.rs

```diff
@@ -111,25 +111,34 @@ impl BlockImporter for BlockImporterAdapter {
     }
 }
 
-const TIME_UNTIL_SYNCED: Duration = Duration::from_secs(60);
-
-#[async_trait::async_trait]
-impl SyncPort for crate::service::adapters::SyncAdapter {
-    async fn sync_with_peers(&mut self) -> anyhow::Result<()> {
-        // todo: connect to N amount of peers first!
-
-        'outer: loop {
-            tokio::select! {
-                _ = self.block_rx.recv() => {
-                    // keep receiving them blocks
-                }
-                _ = tokio::time::sleep(TIME_UNTIL_SYNCED) => {
-                    // time expired we are synced!
-                    break 'outer;
-                }
-            }
+#[cfg(feature = "p2p")]
+impl P2pPort for P2PAdapter {
+    fn reserved_peers_count(&self) -> Receiver<usize> {
```

Comment:
We try not to use `tokio` types in the public API. It should return a stream as we do it in other cases.

Also, you don't need to return the closed stream if p2p is disabled. It should be an infinite stream like `pending`.